src/script/interpreter.cpp

```diff
@@ -1073,22 +1076,14 @@ class CTransactionSignatureSerializer {
 
         // Serialize vpour
         if (txTo.nVersion >= 2) {
-            // TODO:
             //
             // SIGHASH_* functions will hash portions of
             // the transaction for use in signatures. This
-            // keeps the pour cryptographically bound to
-            // the transaction from the perspective of the
-            // inputs (but not from the perspective of the
-            // pour).
-            //
-            // This must be rectified in the future.
-            // See zcash/#529
+            // keeps the JoinSplit cryptographically bound
+            // to the transaction.
             //
-            // It will be necessary to change this API to
-            // be abstract over whether an input script is
-            // being skipped or a pour is being skipped.
             ::Serialize(s, txTo.vpour, nType, nVersion);
+            ::Serialize(s, txTo.joinSplitPubKey, nType, nVersion);
```

Comment:
Yes, this isn't finished. Both ``joinSplitPubKey`` and ``joinSplitSig`` should only be serialised when there is at least one JoinSplit.